src/System.Device.Ports/System.Device.Ports.SerialPort/SerialPort.Settings.cs

```diff
@@ -0,0 +1,728 @@
+            get
+            {
+                if (!_isOpen)
+                {
+                    throw new InvalidOperationException(Strings.Port_not_open);
+                }
+
+                return GetCtsHolding();
+            }
+        }
+
+        /// <summary>
+        /// Gets the state of the Clear-to-Send line.
+        /// </summary>
+        protected internal abstract int GetCtsHolding();
+
+        /// <summary>
+        /// Gets or sets a value indicating whether null bytes are ignored when
+        /// transmitted between the port and the receive buffer.
+        /// </summary>
+        public bool DiscardNull
+        {
+            get => _discardNull;
+            set
+            {
+                /*
+                if (value == _discardNull)
+                {
+                    return;
+                }
+                */
+
+                _discardNull = value;
+                SetDiscardNull(_discardNull);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether null bytes are ignored when
+        /// transmitted between the port and the receive buffer.
+        /// </summary>
+        /// <param name="value">true if null bytes are ignored; otherwise false. The default is false.</param>
+        protected internal abstract void SetDiscardNull(bool value);
+
+        /// <summary>
+        /// Gets the state of the Data Set Ready (DSR) signal.
+        /// </summary>
+        public int DsrHolding
+        {
+            get
+            {
+                if (!_isOpen)
+                {
+                    throw new InvalidOperationException(Strings.Port_not_open);
+                }
+
+                return GetDsrHolding();
+            }
+        }
+
+        /// <summary>
+        /// Gets the state of the Data Set Ready (DSR) signal.
+        /// </summary>
+        protected internal abstract int GetDsrHolding();
+
+        /// <summary>
+        /// Gets or sets a value that enables the Data Terminal Ready (DTR) signal during serial communication.
+        /// </summary>
+        public bool DtrEnable
+        {
+            get => _dtrEnable;
+            set
+            {
+                /*
+                if (value == _dtrEnable)
+                {
+                    return;
+                }
+                */
+
+                _dtrEnable = value;
+                SetDtrEnable(_dtrEnable);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value that enables the Data Terminal Ready (DTR) signal during serial communication.
+        /// </summary>
+        /// <param name="value">true to enable Data Terminal Ready (DTR); otherwise, false. The default is false.</param>
+        protected internal abstract void SetDtrEnable(bool value);
+
+        /// <summary>
+        /// Gets or sets the byte encoding for pre- and post-transmission conversion of text.
+        /// </summary>
+        public Encoding Encoding
```

Comment:
I'd prefer we didn't depend on encoding (i.e. remove all string related helper methods)

Reply:
As I wrote in my comment after the last push, I am willing to remove any read/write method from the SerialPort class.
Instead, I plan to add a sort of "plugin" (based on an interface) that allows to customize the typology of reads/writes.
For example Laurent asked to have separate streams for the read and write channels.